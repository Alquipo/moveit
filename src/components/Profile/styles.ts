import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;

    align-items: center;
    color: ${theme.colors.text};
    transition: color 0.3s;

    div {
      margin-left: 1.5rem;

      strong {
        font-size: 1.5rem;
        font-weight: 600;
        color: ${theme.colors.title};
        transition: color 0.3s;
      }

      p {
        font-size: 1rem;
        margin-top: 0.5rem;

        img {
          margin-right: 0.5rem;
        }
      }
    }
  `}
`
export const ProfileImage = styled.img`
  width: 5.5rem;
  height: 5.5rem;
  border-radius: 50%;
`
