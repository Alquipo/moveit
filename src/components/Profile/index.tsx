import { ChallengesContext } from 'contexts/ChallengesContext'
import { useContext } from 'react'
import Link from 'next/link'
import { LogOut } from '@styled-icons/ionicons-outline/LogOut'
import * as S from './styles'
import { useAuth } from 'contexts/AuthContext'

const Profile = () => {
  const { level } = useContext(ChallengesContext)
  const { userData } = useAuth()

  const { logout } = useAuth()

  return (
    <S.Wrapper>
      <S.ProfileImage src={userData?.photo} alt={userData?.name} />

      <div>
        <strong>{userData?.name}</strong>
        <p>
          <img src="icons/level.svg" alt="Level" />
          Level {level}
        </p>
      </div>

      <Link href="/">
        <LogOut size={40} role="button" onClick={logout} />
      </Link>
    </S.Wrapper>
  )
}

export default Profile
