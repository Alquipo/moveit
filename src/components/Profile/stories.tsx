import { Story, Meta } from '@storybook/react/types-6-0'
import Profile from '.'

export default {
  title: 'Profile',
  component: Profile
} as Meta

export const Default: Story = (props) => <Profile {...props} />
