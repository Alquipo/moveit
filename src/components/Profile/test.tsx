import { render, screen } from '@testing-library/react'

import Profile from '.'

describe('<Profile />', () => {
  it('should render the heading', () => {
    const { container } = render(<Profile />)

    expect(screen.getByRole('heading', { name: /Profile/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
