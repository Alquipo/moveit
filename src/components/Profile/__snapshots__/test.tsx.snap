// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render the Profile correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #EEEEEE;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 div {
  margin-left: 1.5rem;
}

.c0 div strong {
  font-size: 1.5rem;
  font-weight: 600;
  color: #FFFFFF;
}

.c0 div p {
  font-size: 1rem;
  margin-top: 0.5rem;
}

.c0 div p img {
  margin-right: 0.5rem;
}

.c1 {
  width: 5.5rem;
  height: 5.5rem;
  border-radius: 50%;
}

<div
  class="c0"
>
  <img
    alt="Alquipo Neto"
    class="c1"
    src="https://github.com/alquipo.png"
  />
  <div>
    <strong>
      Alquipo Neto
    </strong>
    <p>
      <img
        alt="Level"
        src="icons/level.svg"
      />
      Level 1
    </p>
  </div>
</div>
`;
